cmd/crates/soroban-spec-tools/src/contract.rs

```diff
@@ -52,7 +51,13 @@ impl Spec {
                     "contractspecv0" => &mut spec,
                     _ => continue,
                 };
-                *out = Some(section.data());
+
+                if let Some(existing_data) = out.as_mut() {
+                    let combined_data = [existing_data, section.data()].concat();
+                    *out = Some(Box::leak(combined_data.into_boxed_slice()));
```

Comment:
Does this mean if a .wasm has meta with field a in it, that the new custom section written will have the first custom section copied into it then append onto?

That'll mean the meta for a will appear twice, in both custom sections.

It should be okay to not join them, and just append the new custom section with the additional fields.

Reply:
🤔 I may be misunderstanding how this `Spec` struct is being used - I'm interpreting it as a reader here, since it parses the wasm bytes into an easier-to-manage type.

Without joining the existing data with the new custom section here, I'm seeing that even though the new custom sections were added to the wasm properly, when I print out the contract's meta (via `stellar contract info meta`) only the last meta passed in on build was being displayed.

For example, if I run:
```
stellar contract build --metadata new-meta-key=new-meta-value \ 
    --metadata another-new-meta-key=another-new-meta-value
```
I would expect that my contract should have the following metadata:
```
- Description: A test hello world contract (added with contractmeta macro in the contract)
- rsver: 1.81.0 (builtin meta)
- rssdkver: 21.7.3#1f81... (builtin meta) 
- new-meta-key: new-meta-value (added on build)
- another-new-meta-key: another-new-meta-value (added on build)
```

But when I run `stellar contract info meta`, I only see this:
```
Contract meta:
 • another-new-meta-key: another-new-meta-value
```